pkg/compactor/block_upload.go

```diff
@@ -34,10 +35,11 @@ import (
 )
 
 const (
-	uploadingMetaFilename       = "uploading-meta.json" // Name of the file that stores a block's meta file while it's being uploaded
-	validationFilename          = "validation.json"     // Name of the file that stores a heartbeat time and possibly an error message
-	validationHeartbeatInterval = 1 * time.Minute       // Duration of time between heartbeats of an in-progress block upload validation
-	validationHeartbeatTimeout  = 5 * time.Minute       // Maximum duration of time to wait until a validation is able to be restarted
+	uploadingMetaFilename       = "uploading-meta.json"  // Name of the file that stores a block's meta file while it's being uploaded
+	validationFilename          = "validation.json"      // Name of the file that stores a heartbeat time and possibly an error message
+	validationHeartbeatInterval = 1 * time.Minute        // Duration of time between heartbeats of an in-progress block upload validation
+	validationHeartbeatTimeout  = 5 * time.Minute        // Maximum duration of time to wait until a validation is able to be restarted
+	maximumMetaSizeBytes        = int64(1 * 1024 * 1024) // 1 MiB, maximum allowed size of an uploaded block's meta.json file
```

Comment:
We don't. I defined the type explicitly because it's only passed to `http.MaxBytesReader` which takes an `int64`. I changed it to keep it untyped if that stood out as odd though.